kernel/src/memory/memory_init.rs

```diff
@@ -17,7 +17,7 @@ pub fn init(boot_info: &'static BootInfo) {
             .expect("physical memory mapping not set"),
     );
     unsafe { page_table::init(pmo) };
-    let mut frame_allocator = unsafe { FullFrameAllocator::init(&boot_info.memory_regions) };
+    let mut frame_allocator = BitmapFrameAllocator::init(&boot_info);
```

Comment:
what clippy says. if you have a function `fn foo(x: &bar)` that is called like `fn baz(x: &bar) { foo(x); }` you dont need to reference it (also works if `&mut bar` and for `&'static bar`)